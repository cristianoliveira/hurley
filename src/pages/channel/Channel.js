import React, { Component } from 'react';
import { Link } from 'react-router';
import Helmet from 'react-helmet';

import helpers from 'app/scripts/helpers'

import 'app/styles/channel.sass'
import ChannelEpisodes from 'app/components/ChannelEpisodes'


class Channel extends Component {
  renderCategories() {
    return this.props.channel.categories.map( (i,n) => (
      <Link key={n} to={`/category/${i.slug}`}>
        <i className={`fa fa-${i.icon}`}></i> {helpers.translate(i.title)}
      </Link>
    ));
  }

  render() {
    if (!this.props.channel) {
      return null;
    }

    const {
      description,
      image_url,
      title,
    } = this.props.channel;

    const metaTitle = this.props.channel ? `| ${title}` : ``

    return (
      <div className="feedcast__channelPage">
        <Helmet
          title={`Feedcast ${metaTitle}`}
          meta={[
            {property: 'og:title',
            content: `Feedcast ${metaTitle}`},
          ]} />
        <div className="feedcast__channelInfo">
          <div className="feedcast__channelInfo-header">
            <img className="feedcast__channelInfo-img" alt={title} src={image_url}/>
            <h1 className="feedcast__channelInfo-title">{title}</h1>
          </div>
          <div className="feedcast__channelInfo-body">
            <p className="feedcast__channelInfo-description">{description}</p>
            <div className="feedcast__channelInfo-categories">
              { this.renderCategories.bind(this) }
            </div>
          </div>
        </div>
        <ChannelEpisodes {...this.props} />
      </div>
    );
  }
}

export default Channel;
