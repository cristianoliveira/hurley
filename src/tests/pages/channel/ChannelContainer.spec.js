import React from 'react';
import renderer from 'react-test-renderer';
import { withStore } from 'app/tests/helpers';

import Channel from 'app/pages/channel';

describe('Channel', () => {
  const state = { channels: {} };

  describe('when has no data', () => {
    it("renders the no episode", () => {
      const component = renderer.create(withStore(<Channel />, state));
      expect(component).toMatchSnapshot();
    })
  });

  describe('when is fetching', () => {
    const state = { channels: { isFetching: true } };

    it("renders the loader", () => {
      const component = renderer.create(withStore(<Channel />, state));
      expect(component).toMatchSnapshot();
    })
  });

  describe('when has a list of channels', () => {
    const channels = {
      channels: [
        {
          uuid: "asdasdadasda",
          slug: "pensando-rpg",
          title: "Pensando RPG",
          description: "",
          feed_url: "http://feeds.soundcloud.com/users/soundcloud:users:257525576/sounds.rss",
          image_url: "https://img.feedcast.io/http%3A%2F%2Fi.imgur.com%2FmYRfJk7.png/300/300",
          listed: true,
          synchronization_status: "success",
          synchronization_status_message: "",
          categories: [],
        }
      ],
      page: 1,
      perPage: 130,
      total: 1000,
    };

    const state = { channels };

    it('render the channels', () => {
      const component = renderer.create(withStore(withStore(<Channel />, state)));
      expect(component).toMatchSnapshot();
    });
  });
})
