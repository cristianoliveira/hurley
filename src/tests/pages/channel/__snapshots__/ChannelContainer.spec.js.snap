// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel when has a list of channels render the channels 1`] = `
<div
  className="feedcast__loader"
>
  <div
    className="feedcast__loader--top"
  />
</div>
`;

exports[`Channel when has no data renders the no episode 1`] = `
<div
  className="feedcast__loader"
>
  <div
    className="feedcast__loader--top"
  />
</div>
`;

exports[`Channel when is fetching renders the loader 1`] = `
<div
  className="feedcast__loader"
>
  <div
    className="feedcast__loader--top"
  />
</div>
`;
